GameOfLife.java

```diff
@@ -63,16 +69,56 @@ public static int[][] read(String fileName) {
 		int rows = Integer.parseInt(in.readLine());
 		int cols = Integer.parseInt(in.readLine());
 		int[][] board = new int[rows + 2][cols + 2];
+		char z = ' ';
+		
+			for (int i = 1 ; i<board.length; i++)
+			{	
+				String text = in.readLine();
+				if (text != null && !text.isEmpty())
+				{
+				for (int j= 1; j<board[1].length; j++ )
+				{
+					if (j -1 < text.length())
+					{
+						z = text.charAt(j-1);
+					
+						if (z == 'x')
+						{
+							board[i][j] = 1;
+						}
+
+						else 
+						{
+						board [i][j] = 0;
+						}
+
+					}
+				}
+				}
+		}
+
 		//// Replace the following statement with your code.
-		return null;
+		return board;
 	}
 	
 	// Creates a new board from the given board, using the rules of the game.
 	// Uses the cellValue(board,i,j) function to compute the value of each 
 	// cell in the new board. Returns the new board.
-	public static int[][] evolve(int[][] board) {
+	public static int[][] evolve(int[][] board)
+	{
 		//// Replace the following statement with your code.
-		return null;
+
+		int [][] newBoard = new int [board.length][board[0].length];
+
+		for (int i=0 ;i<board.length-1; i++)
```

Comment:
When `i=0` you will iterate indices out of boundaries at `count` method.
```suggestion
		for (int i=1 ;i<board.length-1; i++)
```